composer/models/huggingface.py

```diff
@@ -403,6 +421,40 @@ def get_metadata(self):
                     }
         return {'model': model_output, 'tokenizer': tokenizer_output}
 
+    def generate(self,
+                 input_ids: torch.Tensor,
+                 num_beams: int = 1,
+                 do_sample: bool = False,
+                 max_new_tokens: int = 20,
```

Comment:
why these but not e.g. top_p?

Reply:
I assume we can pass that with kwargs? I am just curious why these are explicitly not in kwargs

Reply:
Yeah this is a fair question. I called out the really important ones explicitly for clarity...but there isn't any particular reason to do this. Open to putting everything in kwargs. Max length is extremely critical in order for this to work with fsdp, but the others are not.

Reply:
`num_beams` feels less important to me than `top_p`, since nucleus sampling is considered better lately. And if we expose `num_beams` I feel like we should include `no_repeat_ngram_size`.

I think I am in favor of explicitly exposing all the generation parameters. One benefit to this is that we can **check if the user sets `penalty_alpha` to a non-falsey value; if so I am pretty sure we need to error, since I don't think we support contrastive decoding** currently (we will need to conform our forward signature to huggingface standards; we need to support [output_hidden_states](https://github.com/huggingface/transformers/blob/v4.27.1/src/transformers/generation/utils.py#L1834) and [passing around the KV cache](https://github.com/huggingface/transformers/blob/v4.27.1/src/transformers/generation/utils.py#L1858-L1862)

Reply:
There are kind of a lot of generate parameters... https://huggingface.co/docs/transformers/v4.27.1/en/main_classes/text_generation#transformers.GenerationConfig. I don't think _we_ (being `HuggingFaceModel`) need to error on any generation parameters, because the call to generate happens on the underlying HF model, and presumably HF errors if you do stuff that is not allowed. We (being `MosaicGPT`) may need some error handling for things that aren't supported yet in the `MosaicGPT` model. Also, if there are any args that we need to check here for some reason, we can just pop them off of. Ok, what if I just don't specially expose any args, and defer to the HF docs?

Reply:
ok i dropped all the explicit args and left a note in the docstring about where to find them in the HF docs.